src/tests/Microsoft.PowerFx.Interpreter.Tests/Scenarios/CustomRecordTypeWithUndefinedFields.cs

```diff
@@ -0,0 +1,201 @@
+namespace Microsoft.PowerFx.Interpreter.Tests
+{
+    // Demonstrate the ability to override the behavior when accessing undefined fields.
+    public class CustomRecordTypeWithUndefinedFields : PowerFxTest
+    {
+        private readonly RecordType _originalRecordType;
+        private readonly RecordType _customRecordType;
+        private readonly TestObj _testObj;
+
+        public CustomRecordTypeWithUndefinedFields()
+        {
+            _originalRecordType = RecordType
+                .Empty()
+                .Add(new NamedFormulaType("prop1", FormulaType.Number))
+                .Add(new NamedFormulaType("prop2", FormulaType.Number))
+                .Add(new NamedFormulaType("prop3", FormulaType.Number));
+
+            _customRecordType = new TreatMissingFieldAsUntypedRecordType(_originalRecordType);
+            _testObj = new TestObj
+            {
+                Properties = new Dictionary<string, object>
+                {
+                    { "prop1",  1 },
+                    { "prop2",  "hello" },
+                    { "prop_not_defined_in_type",  3 },
+                }
+            };
+        }
+
+        [Theory]
+        [InlineData("obj.prop1", 1.0)]
+        [InlineData("IsError(obj.prop2)", true)] // type mismatch should fail at runtime
+        [InlineData("IsBlank(obj.missing)", "#exception")] // type checking fails when referencing undefined type.
+        [InlineData("IsBlank(obj.missing.missing)", "#exception")]
+        [InlineData("IsBlank(obj.prop_not_defined_in_type)", "#exception")]
+        [InlineData("IsError(obj.missing + 1)", "#exception")]
+        public async Task TestOriginalRecordType(string expr, object expected)
+        {
+            var engine = new RecalcEngine();
+            var symbolTable = new SymbolTable();
+            var symbolValues = new SymbolValues();
+
+            symbolTable.AddVariable("obj", _originalRecordType);
+            symbolValues.Add("obj", new TestCustomRecordValue(_testObj, _originalRecordType));
+
+            try
+            {
+                var result = await engine.EvalAsync(expr, default, null, symbolTable, symbolValues).ConfigureAwait(false);
+                Assert.Equal(expected, result.ToObject());
+            }
+            catch (InvalidOperationException ex)
+            {
+                var str = Assert.IsType<string>(expected);
+                Assert.Equal("#exception", str);
+                Console.WriteLine(ex.Message);
+            }
+        }
+
+        [Theory]
+        [InlineData("obj.prop1", 1.0)]
+        [InlineData("IsError(obj.prop2)", true)] // type mismatch should fail at runtime.
+        [InlineData("IsBlank(obj.missing)", true)] // fields not defined in RecordType is treated as Blank.
+        [InlineData("IsBlank(obj.missing.missing)", true)]
+        [InlineData("IsBlank(obj.prop_not_defined_in_type)", true)] // fields not defined in RecordType is always Blank, regardless of whether it is defined in actual data.
```

Comment:
note: this is the behavior defined on lines 117-121, not innate to the system. 
A different implementation could be returning an UntypedObjectValue for these.

Reply:
yes, I explicitly define that because I want to able to have a "blank if not exist" behavior in the expressions.